### PR TITLE
Placeholder support in workflows.

### PR DESCRIPTION
**Pull request by andreas-gruenwald**
Support for placeholders in workflows. By using placeholders, it is possible, to reuse an existing workflow and reapply places and transitions onto a new workflow.

Example:
```
pimcore:
  workflows:
    product_workflow_local_de: &local_workflow_reference
        placeholders:
          '%%stateFieldName%%': 'translationStateDe'
          '%%locale%%': 'de'
        enabled: true
        label: "Local Release Workflow"
        type: "state_machine"
        supports: &local_workflow_supports_ref
          - \Pimcore\Model\DataObject\Product
        initial_place: "local_300_basic_data_pending"
        marking_store: &local_workflow_marking_store_ref
          type: single_state
          arguments:
            -
              %%stateFieldName%%
        places: &local_workflow_places_ref

            100_new:
              label: "Basic Data Pending from Global Workflow (%%locale%%)"
              visibleInHeader: true
              color:  '#cddc39'
              colorInverted: false
              permissions:
                - publish: true

            local_300_basic_data_pending:
              label: "Pending"
              visibleInHeader: true
              color:  '#cddc39'
              colorInverted: false
              permissions:
                - publish: true

             ...
             
        transitions: &local_workflow_transitions_ref
            to_published:
                to: local_400_published
                from: [local_700_locked]
                guard: is_fully_authenticated() and (has_role('ROLE_PIMCORE_ADMIN') or has_role('ROLE_RECIPE_WORKFLOW_FINISH_DEVELOPMENT'))
                options:
                  label: 'Set to local_400_published'

            ....
```
Example cloned workflows:
```
    product_workflow_local_en:
        placeholders:
          '%%stateFieldName%%': 'translationStateEn'
          '%%locale%%': 'en'
          '%%flag%%': 'gb'
        enabled: true
        label: "Local Release Workflow"
        type: "state_machine"
        supports:
          <<: *local_workflow_supports_ref
        initial_place: "local_300_basic_data_pending"
        marking_store:
          <<: *local_workflow_marking_store_ref
        places:
          <<: *local_workflow_places_ref
        transitions:
          <<: *local_workflow_transitions_ref

    product_workflow_local_hr:
      placeholders:
        '%%stateFieldName%%': "translationStateHr"
        '%%locale%%': "hr"
        '%%flag%%': 'hr'
      enabled: true
      label: "Local Release Workflow"
      type: "state_machine"
      supports:
        <<: *local_workflow_supports_ref
      initial_place: "local_300_basic_data_pending"
      marking_store:
        <<: *local_workflow_marking_store_ref
      places:
        <<: *local_workflow_places_ref
      transitions:
        <<: *local_workflow_transitions_ref

    product_workflow_local_ro:
      placeholders:
        '%%stateFieldName%%': "translationStateRo"
        '%%locale%%': "ro"
        '%%flag%%': 'ro'
      enabled: true
      label: "Local Release Workflow"
      type: "state_machine"
      supports:
        <<: *local_workflow_supports_ref
      initial_place: "local_300_basic_data_pending"
      marking_store:
        <<: *local_workflow_marking_store_ref
      places:
        <<: *local_workflow_places_ref
      transitions:
        <<: *local_workflow_transitions_ref
```
...

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
What about also allow expression to things like `object.getName` ?

File: bundles/CoreBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 namespace Pimcore\Bundle\CoreBundle\DependencyInjection;
 
+use Pimcore\Bundle\EcommerceFrameworkBundle\DependencyInjection\Config\Processor\PlaceholderProcessor;
```
Comment (fashxp) [inline bundles/CoreBundle/DependencyInjection/Configuration.php]: hmm, could you move the placeholderprocesser into corebundle and reference it in ecommerce bundle. 
feels wrong the other way round.

File: lib/Workflow/Place/StatusInfo.php [hunk 2]
```diff
@@ -111,7 +111,8 @@ private function getAllPlaces($subject, bool $visibleInHeaderOnly = false, strin
          * @var PlaceConfig $place
          */
         foreach ($places as $place) {
-            $uniquePlaces[$place->getPlace()] = $place;
+            $uniqueLabel = $place->getLabel().'__'.$place->getPlace();
```
Comment (fashxp) [inline lib/Workflow/Place/StatusInfo.php]: should that be really part of the PR? 
seems like something different

**Comment by fashxp** [comment]
nice addition!!

**Comment by fashxp** [comment]
@dpfaffenbauer 
> What about also allow expression to things like `object.getName` ?

What do you mean with that?

File: bundles/CoreBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (andreas-gruenwald) [reply bundles/CoreBundle/DependencyInjection/Configuration.php]: Update done.

File: lib/Workflow/Place/StatusInfo.php (hunk 2 above)
Comment (andreas-gruenwald) [reply lib/Workflow/Place/StatusInfo.php]: Removed again.

File: bundles/CoreBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (andreas-gruenwald) [reply bundles/CoreBundle/DependencyInjection/Configuration.php]: - Placeholderprocessor moved.
- Documentation extended.

**Pull request merged**